src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -146,6 +146,19 @@ public static TypeSymbol GetNullableUnderlyingType(this TypeSymbol type)
             return type.GetNullableUnderlyingTypeWithAnnotations().Type;
         }
 
+        public static bool IsSystemNullable(this TypeSymbol? type, [NotNullWhen(true)] out TypeSymbol? underlyingType)
+        {
+            if (type is NamedTypeSymbol nt
+                && nt.OriginalDefinition.GetSpecialTypeSafe() == SpecialType.System_Nullable_T)
```

Comment:
what's the difference between this and IsNullableType above?  specifically, why GetSpecialTypeSafe() versus .SpecialType?

Reply:
should IsNullableType defer to IsSystemNullable so everything is consistent?

Reply:
why do you need to check if its' a NamedType if you're already going use the SpecialTYpe?

Reply:
> what's the difference between this and IsNullableType above?

The difference is that we're returning the underlying type.

>  specifically, why GetSpecialTypeSafe() versus .SpecialType?

This is a refactoring of existing code. `GetSpecialTypeSafe` is just `SpecialType` plus a null check. That said, we don't need `GetSpecialTypeSafe()` here since `OriginalDefinition` cannot be null.

> should IsNullableType defer to IsSystemNullable so everything is consistent?

I don't see a need.

> why do you need to check if its' a NamedType if you're already going use the SpecialTYpe?

We can only access `TypeArgumentsWithAnnotationsNoUseSiteDiagnostics` on a `NamedTypeSymbol`.

Reply:
My basic point is that the these functions are virtually the same, except for the convenience of getting the type argument. As such, keeping them as similar as possible (naming, impl, etc.) seems like the best thing to do. Having them be subtly different just seems like a way for bugs to creep in. 

Wrt the named type check. It seems better to do the special type check first, then just do a cast afterwards.